reference/store.go

```diff
@@ -54,15 +54,19 @@ type repository map[string]digest.Digest
 
 type lexicalRefs []reference.Named
 
-func (a lexicalRefs) Len() int           { return len(a) }
-func (a lexicalRefs) Swap(i, j int)      { a[i], a[j] = a[j], a[i] }
-func (a lexicalRefs) Less(i, j int) bool { return a[i].String() < a[j].String() }
+func (a lexicalRefs) Len() int      { return len(a) }
+func (a lexicalRefs) Swap(i, j int) { a[i], a[j] = a[j], a[i] }
+func (a lexicalRefs) Less(i, j int) bool {
+	return reference.FamiliarString(a[i]) < reference.FamiliarString(a[j])
+}
 
 type lexicalAssociations []Association
 
-func (a lexicalAssociations) Len() int           { return len(a) }
-func (a lexicalAssociations) Swap(i, j int)      { a[i], a[j] = a[j], a[i] }
-func (a lexicalAssociations) Less(i, j int) bool { return a[i].Ref.String() < a[j].Ref.String() }
+func (a lexicalAssociations) Len() int      { return len(a) }
+func (a lexicalAssociations) Swap(i, j int) { a[i], a[j] = a[j], a[i] }
+func (a lexicalAssociations) Less(i, j int) bool {
+	return reference.FamiliarString(a[i].Ref) < reference.FamiliarString(a[j].Ref)
```

Comment:
Actually, now that I think about it, this is terrible, because every sort comparison has to allocate new strings.

I believe the reason this sorting was added was so that unit tests could more easily check the results of `References` and `ReferencesByName`. Maybe we should change this back to sorting on the normalized name, even though this is a slight change in behavior.